src/components/control/select.jsx

```diff
@@ -0,0 +1,295 @@
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../icon/icon.jsx';
+import IconButton from '../icon-button/icon-button.jsx';
+import iconClear from '../../svgs/clear.svg';
+import iconCaretDown from '../../svgs/caret-down.svg';
+import iconCaretDownDisabled from '../../svgs/caret-down-disabled.svg';
+import cautionSvg from '../../svgs/caution.svg';
+import Control from '../control/control.jsx';
+import Listbox from '../listbox/listbox.jsx';
+import NoOptions from '../no-options/no-options.jsx';
+import styles from '../select/select.css';
+import useValidation from '../../hooks/use-validation.jsx';
+import useDropdownClose from '../../hooks/use-dropdown-close.js';
+import useMounted from '../../hooks/use-mounted.js';
+import useForwardedRef from '../../hooks/use-forwarded-ref.js';
+
+const Select = forwardRef(function Select(props, ref) {
+  const [showOptions, setShowOptions] = useState(false);
+  const [value, setValue] = useState(props.value === undefined ? null : props.value);
+  const [hasBeenBlurred, setBeenBlurred] = useState(false);
+  const [searchValue, setSearchValue] = useState(undefined);
+  const mounted = useMounted();
+  const selfRef = useForwardedRef(ref);
+
+  const defaultValue = value ? props.displayValueEvaluator(value) : '';
+
+  const wrapperRef = useForwardedRef(props.wrapperRef);
+  const handleDropdownClose = () => {
+    setShowOptions(false);
+    setSearchValue(defaultValue);
+  };
+  useDropdownClose(wrapperRef, showOptions, handleDropdownClose);
+
+  const refs = {
+    listbox: useRef(),
+    input: useRef(),
+  };
+
+  const [validationMessage, validate] = useValidation(props.validationMessage, refs.input);
```

Comment:
I feel like I vaguely recall being told not to use this hook; are we okay with using it here since this is mostly copy/paste refactoring?

Reply:
Yeah, I do want to replace it soon too. Thanks for the reminder